.github/workflows/advanced.yml

```diff
@@ -122,7 +122,7 @@ env:
 
 jobs:
   build-toolchain:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.arch == 'aarch64' && 'ubuntu-24.04-arm' || 'ubuntu-24.04' }}
```

Comment:
I would suggest to define 
UBUNTU_LATEST_ARM and use ubuntu-latest for x64 for now. Most probably 
ubuntu-latest-arm might be defined later, otherwise it is easier to update UBUNTU_LATEST_ARM.

Reply:
Let me try. IIRC, environment variables cannot be used for `runs-on:`...